x-pack/plugins/remote_clusters/public/application/sections/components/remote_cluster_form/remote_cluster_form.js

```diff
@@ -296,6 +316,187 @@ export class RemoteClusterForm extends Component {
+              defaultMessage="The address used for all remote connections."
+            />
+          }
+          isInvalid={Boolean(areErrorsVisible && errorProxyAddress)}
+          error={errorProxyAddress}
+          fullWidth
+        >
+          <EuiFieldText
+            value={proxyAddress}
+            placeholder={i18n.translate(
+              'xpack.remoteClusters.remoteClusterForm.fieldProxyAddressPlaceholder',
+              {
+                defaultMessage: 'host:port',
+              }
+            )}
+            onChange={e => this.onFieldsChange({ proxyAddress: e.target.value })}
+            isInvalid={Boolean(areErrorsVisible && errorProxyAddress)}
+            data-test-subj="remoteClusterFormProxyAddressInput"
+            fullWidth
+          />
+        </EuiFormRow>
+
+        <EuiFormRow
+          data-test-subj="remoteClusterFormProxySocketConnectionsFormRow"
+          label={
+            <FormattedMessage
+              id="xpack.remoteClusters.remoteClusterForm.fieldProxySocketConnectionsLabel"
+              defaultMessage="Socket connections"
+            />
+          }
+          helpText={
+            <FormattedMessage
+              id="xpack.remoteClusters.remoteClusterForm.fieldSocketConnectionsHelpText"
+              defaultMessage="The number of socket connections to open per remote cluster."
+            />
+          }
+          fullWidth
+        >
+          <EuiFieldNumber
+            value={proxySocketConnections || ''}
+            onChange={e =>
+              this.onFieldsChange({ proxySocketConnections: Number(e.target.value) || null })
+            }
+            fullWidth
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          data-test-subj="remoteClusterFormServerNameFormRow"
+          label={
+            <FormattedMessage
+              id="xpack.remoteClusters.remoteClusterForm.fieldServerNameLabel"
+              defaultMessage="Server name"
+            />
+          }
+          helpText={
+            <FormattedMessage
+              id="xpack.remoteClusters.remoteClusterForm.fieldServerNameHelpText"
+              defaultMessage="An optional hostname string which will be sent in the server_name field of the TLS Server Name Indication extension if TLS is enabled."
+            />
+          }
+          fullWidth
+        >
+          <EuiFieldText
+            value={serverName}
+            onChange={e => this.onFieldsChange({ serverName: e.target.value })}
+            fullWidth
+          />
+        </EuiFormRow>
+      </>
+    );
+  }
+
+  renderMode() {
+    const {
+      fields: { mode },
+    } = this.state;
+
+    return (
+      <EuiDescribedFormGroup
+        title={
+          <EuiTitle size="s">
+            <h2>
+              <FormattedMessage
+                id="xpack.remoteClusters.remoteClusterForm.sectionModeTitle"
+                defaultMessage="Connection mode settings"
```

Comment:
I'll defer to @gchaps if she disagrees, but I think this can be simplified to just "Connection mode", since technically everything in this form is a setting.